### ISSUE TITLE
Preparing to archive Krustlet: comments?

### ISSUE THREAD

**Issue by squillace** [deislabs/krustlet#758]
Dear all, including some core folks like @thomastaylor312, @radu-matei, @technosophos, @flavio, @kflansburg, @ereslibre and others, I am going to file a PR with the CNCF TOC to gain permission to archive the Krustlet project. Before I do, however, I wanted to invite comment about the code involved, in particular the oras-rs and krator code that produces those crates.

How do you all feel about this archive proposal? And does anyone want the crates to live on somewhere else?

Cheers, Ralph

**Comment by ereslibre** [comment]
> How do you all feel about this archive proposal?

+1 from my side. It has been a fun ride, amazing project and team that made it possible. Thanks @squillace for reaching out.

> And does anyone want the crates to live on somewhere else?

No preference from my side. Count me in if there's any work needed in those crates, willing to contribute.

**Comment by thomastaylor312** [comment]
My only concern is around the oci-distribution crate, which is actively used and is under the krustlet org

**Comment by bacongobbler** [comment]
I know @thomastaylor312 has been actively developing the oci-distribution crate. v0.11.0 [was released last week](https://github.com/krustlet/oci-distribution/releases/tag/v0.11.0). wasmcloud and spin both utilize that crate in production today (with the latter using a fork to pin the dependency, which we could probably bump to 0.11).

https://github.com/wasmCloud/wasmCloud/blob/c354c570308bfd6ab69cc9f7eadf293b74ef19a8/Cargo.toml#L218

https://github.com/fermyon/spin/blob/9f1d9dbc43ae0bde8ffe3729ac9b545b7d571e15/crates/oci/Cargo.toml#L20

**Comment by bacongobbler** [comment]
I'm pretty sure Krator is okay to archive. There's an email chain dated back to Jan 5 2024 between Amye, Taylor, Kevin and myself all in agreement to submit a PR to the toc repo to move it to archive. We just haven't done that yet

**Comment by kflansburg** [comment]
> I'm pretty sure Krator is okay to archive. There's an email chain dated back to Jan 5 2024 between Amye, Taylor, Kevin and myself all in agreement to submit a PR to the toc repo to move it to archive. We just haven't done that yet

I'm fine with this. 

Edit: misread, Krator should already be archived, I submitted it

**Comment by bacongobbler** [comment]
> Krator should already be archived, I submitted it

Ah... The repository itself has not been archived. That's what confused me.

https://github.com/krator-rs/krator

EDIT: I realized I have the power to archive the repo. It is done.

**Comment by bacongobbler** [comment]
> And does anyone want the crates to live on somewhere else?

If the end result here is that we need to fork it and/or choose a non-krustlet location to host this crate, perhaps the [ORAS project](https://github.com/oras-project) would be happy to provide a new home for it, since that is also a CNCF project. Or perhaps the [Open Container Initiative](https://github.com/opencontainers)? Though they're more focused on the spec and Go library rather than maintaining and/or hosting third party client libs.

If neither are suitable new locations for the project, I'd let @thomastaylor312 and @flavio determine its new home - they've been actively contributing and maintaining the oci-distribution project for the past few years.

**Comment by thomastaylor312** [comment]
@flavio You ok if I reach out to ORAS? That seems to be probably the best fit given that it is CNCF

**Comment by flavio** [comment]
@thomastaylor312: I'm happy moving `oci-distribution` under the ORAS umbrella. Go ahead and reach out to them.

Thanks to everybody!

BTW, `oci-distribution` is used in production by [kubewarden](https://kubewarden.io), another CNCF project.

**Comment by thomastaylor312** [comment]
Just a heads up that the ORAS project has started a vote to accept oci-distribution into the project. Their governance requires the vote stay open for 7 days, so we should have a response at that time. I believe based on the agreement in this issue we have done our due diligence from the Krustlet side, but please let me know if there is anything we missed!

**Comment by Toasterson** [comment]
Hey, I just started working on my own kubelet rust implementation for illumos. And I saw this projects kubelet crate was abandoned. I would maintain it anyway fo my own purposes, so I could take it over. It saves me a lot of work already anyway so I'll grab a copy. Wasm is not my goal so if people are more happy that I grab a copy into my own code, I'll simply do that.

**Comment by thomastaylor312** [comment]
@Toasterson is your plan to make it a general purpose thing that would fall under the CNCF? If there is plans to make it bigger, it might be worth keeping around. Either way though, it is Apache 2, so you can fork and start from there if you'd like without any issue!

**Comment by Toasterson** [comment]
My plan would be to keep the kubelet crate general purpose. I don't plan to remove CRI, CNI, CSI or other interfaces to kubelet but I also wont go out of my way to implement Kubelets for machines I wont use it on. I would love to touch ground with the FreeBSD people though, since I am in dev meetings there every once in a while, to see if they want to contribute to this effort. I am not interested in an organization such as the CNCF though. That sounds like much more work than benefit for a crate like this. 

If you have some requirements for taking over the name on crates.io please let me know, I am happy to help keep the supplychain secure but I'm also happy to continue everything under another name.

**Comment by thomastaylor312** [comment]
I don't think we can hand off the name since it is all owned by CNCF. However, like I said there is no problem with forking and continuing under a new project

**Comment by squillace** [comment]
Per https://github.com/bytecodealliance/governance/issues/88 implies we've handled the crate; it seems that the oci-dist part is now renamed and accepted to the ORAS org. I'll be moving on the CNCF TOC PR for archiving.

**Comment by thomastaylor312** [comment]
rust-oci-wasm was different, so you should be good to go now that oci-dist is in ORAS

**Comment by angellk** [comment]
@thomastaylor312 could you please link here the decision to accept OCI-distribution into ORAS

@squillace could you please confirm there are no outstanding code concerns re: Krustlet

**Comment by thomastaylor312** [comment]
@angellk Here is the decision for ORAS: https://github.com/oras-project/community/issues/59
